crates/nu-parser/src/parser.rs

```diff
@@ -450,25 +450,22 @@ fn parse_short_flags(
 ) -> Option<Vec<Flag>> {
     let arg_span = spans[*spans_idx];
 
-    let arg_contents = working_set.get_span_contents(arg_span);
+    let arg_contents = working_set.get_span_contents(arg_span).to_vec();
 
-    if let Ok(arg_contents_uft8_ref) = str::from_utf8(arg_contents) {
+    if let Ok(arg_contents_uft8_ref) = str::from_utf8(&arg_contents) {
         if arg_contents_uft8_ref.starts_with('-') && arg_contents_uft8_ref.len() > 1 {
             let short_flags = &arg_contents_uft8_ref[1..];
+            let num_chars = short_flags.chars().count();
             let mut found_short_flags = vec![];
             let mut unmatched_short_flags = vec![];
-            for short_flag in short_flags.chars().enumerate() {
-                let short_flag_char = short_flag.1;
+            for (i, short_flag) in short_flags.chars().enumerate() {
                 let orig = arg_span;
-                let short_flag_span = Span::new(
-                    orig.start + 1 + short_flag.0,
-                    orig.start + 1 + short_flag.0 + 1,
-                );
-                if let Some(flag) = sig.get_short_flag(short_flag_char) {
-                    // If we require an arg and are in a batch of short flags, error
-                    if !found_short_flags.is_empty() && flag.arg.is_some() {
-                        working_set.error(ParseError::ShortFlagBatchCantTakeArg(short_flag_span));
-                        break;
```

Comment:
I think to be defensive we should still break right?

Reply:
Right. Idk why but I had tried this before and one of my test cases failed with the `break` but it must have been something else.